static/js/components/ShiftPage.jsx

```diff
@@ -71,23 +71,29 @@ const ShiftPage = ({ route }) => {
           Object.keys(updatedShift).length > 0 &&
           Object.keys(currentShift).length > 0
         ) {
-          let usersHaveChanged = false;
-          // check if the users have the same ids, or if they need a replacement when they didnt need one before, and vice versa
-          for (let i = 0; i < updatedShift.shift_users.length; i += 1) {
-            if (
-              updatedShift.shift_users[i].id !==
-                currentShift.shift_users[i].id ||
-              updatedShift.shift_users[i].needs_replacement !==
-                currentShift.shift_users[i].needs_replacement ||
-              updatedShift.shift_users[i].modified !==
-                currentShift.shift_users[i].modified
-            ) {
-              usersHaveChanged = true;
-              break;
-            }
-          }
-          if (usersHaveChanged) {
+          if (
+            updatedShift?.comments?.length !== currentShift?.comments?.length
+          ) {
             dispatch({ type: "skyportal/CURRENT_SHIFT", data: updatedShift });
+          } else {
+            let usersHaveChanged = false;
+            // check if the users have the same ids, or if they need a replacement when they didnt need one before, and vice versa
+            for (let i = 0; i < updatedShift.shift_users.length; i += 1) {
+              if (
+                updatedShift.shift_users[i].id !==
+                  currentShift.shift_users[i].id ||
+                updatedShift.shift_users[i].needs_replacement !==
+                  currentShift.shift_users[i].needs_replacement ||
+                updatedShift.shift_users[i].modified !==
+                  currentShift.shift_users[i].modified
+              ) {
```

Comment:
Could you explain this code a bit more in the comments? also would be nice to have some parentheses in long if conditions.

Reply:
I can explain for Thomas. So basically, when a shift is modified, added, deleted... the list of shift is refreshed. Then, we need to see if the currentShift that has been set before has been updated or deleted. So those are the conditions to check if it was modified (if the users have the same ids, if they need a replacement when they didnt need one before, or simply if the `modified` date has been changed). I think this logic could be simplified and only look at the `modified` value, which should be the result of any modifications anyway.